Core/HTTPSUpgradeStore.swift

```diff
@@ -29,85 +29,106 @@ public protocol HTTPSUpgradeStore {
-        let path = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: ContentBlockerStoreConstants.groupName)
-        return path!.appendingPathComponent("HttpsBloomFilter.bin")
+    private let context = Database.shared.makeContext(concurrencyType: .privateQueueConcurrencyType, name: "HTTPSUpgrade")
+    
+    private var hasBloomFilterData: Bool {
+        return (try? Resource.bloomFilter.checkResourceIsReachable()) ?? false
     }
     
     public func bloomFilter() -> BloomFilterWrapper? {
-        guard hasBloomFilterData else { return nil }
-        var bloomFilter: BloomFilterWrapper?
-        context.performAndWait {
-            if let specification = bloomFilterSpecification() {
-                let entries = specification.totalEntries
-                bloomFilter = BloomFilterWrapper(fromPath: bloomFilterPath.path, withTotalItems: Int32(entries))
-            }
-        }
-        return bloomFilter
+        let storedSpecification = hasBloomFilterData ? bloomFilterSpecification() : loadEmbeddedData()?.specification
+        guard let specification = storedSpecification else { return nil }
+        return BloomFilterWrapper(fromPath: Resource.bloomFilter.path,
+                                  withBitCount: Int32(specification.bitCount),
+                                  andTotalItems: Int32(specification.totalEntries))
     }
-
+    
     public func bloomFilterSpecification() -> HTTPSBloomFilterSpecification? {
         var specification: HTTPSBloomFilterSpecification?
         context.performAndWait {
             let request: NSFetchRequest<HTTPSStoredBloomFilterSpecification> = HTTPSStoredBloomFilterSpecification.fetchRequest()
             guard let result = try? request.execute() else { return }
-            specification = HTTPSBloomFilterSpecification.copy(storedSpecification: result.first)
+            guard let storedSpecification = HTTPSBloomFilterSpecification.copy(storedSpecification: result.first) else { return }
+            guard storedSpecification.bitCount > 0, storedSpecification.totalEntries > 0 else { return }
+            specification = storedSpecification
         }
-        return specification
+        return specification ?? loadEmbeddedData()?.specification
+    }
+    
+    private func loadEmbeddedData() -> EmbeddedBloomData? {
+        os_log("Loading embedded https data")
+        guard let specificationData = try? Data(contentsOf: EmbeddedResource.bloomSpecification) else { return nil }
+        guard let specification = try? JSONDecoder().decode(HTTPSBloomFilterSpecification.self, from: specificationData) else { return nil }
+        guard let bloomData = try? Data(contentsOf: EmbeddedResource.bloomFilter) else { return nil }
+        guard let excludedDomainsData = try? Data(contentsOf: EmbeddedResource.excludedDomains) else { return nil }
+        guard let excludedDomains = try? JSONDecoder().decode(HTTPSExcludedDomains.self, from: excludedDomainsData) else { return nil }
```

Comment:
How do you feel about chaining these declarations under a single `guard`? I find it slightly more readable but am happy either way!

```suggestion
        guard let specificationData = try? Data(contentsOf: EmbeddedResource.bloomSpecification),
              let specification = try? JSONDecoder().decode(HTTPSBloomFilterSpecification.self, from: specificationData),
              let bloomData = try? Data(contentsOf: EmbeddedResource.bloomFilter),
              let excludedDomainsData = try? Data(contentsOf: EmbeddedResource.excludedDomains),
              let excludedDomains = try? JSONDecoder().decode(HTTPSExcludedDomains.self, from: excludedDomainsData)
              else { return nil }
```

Reply:
I expected bundling complex guards would make it harder to read but it looks _a lot_ better. Thanks!